projects/tox4j/src/main/cpp/util/debug_log.cpp

```diff
@@ -53,79 +29,184 @@ register_func (uintptr_t func, char const *name)
   return true;
 }
 
+bool debug_iterate = false;
+
 void
-print_func (uintptr_t func)
+print_func (protolog::JniLogEntry &log_entry, uintptr_t func)
 {
   auto &names = func_names ();
   auto found = names.find (func);
   if (found != names.end ())
-    debug_out << found->second;
+    log_entry.set_name (found->second);
   else
-    debug_out << func;
+    log_entry.set_name (std::to_string (func));
+  if (debug_iterate && log_entry.name () == "tox_iterate")
+    fprintf (stderr, "start: %s\n", log_entry.name ().c_str ());
 }
 
 template<typename Arg>
 void
-print_arg (Arg arg)
+print_arg (protolog::Value &value, Arg arg)
 {
-  debug_out << arg;
+  value.set_sint64 (arg);
 }
 
-template void print_arg<bool              > (bool              );
-template void print_arg<  signed char     > (  signed char     );
-template void print_arg<unsigned char     > (unsigned char     );
-template void print_arg<  signed int      > (  signed int      );
-template void print_arg<unsigned int      > (unsigned int      );
-template void print_arg<  signed long     > (  signed long     );
-template void print_arg<unsigned long     > (unsigned long     );
-template void print_arg<  signed long long> (  signed long long);
-template void print_arg<unsigned long long> (unsigned long long);
+template void print_arg<bool              > (protolog::Value &, bool              );
```

Comment:
Hm, ignore that, apparently it's the same token type as signed/unsigned. ([type-specifier p412](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1124.pdf))